### DISCUSSION TITLE
这是用来检测软件更新的吗？

### DISCUSSION THREAD

**Discussion by xqq524148626** [SpecterShell/Dumplings#21] [Q&A]
这是用来检测软件更新的吗？我看这里面有个华为手机助手的检测，请问它能获得其所有历史版本吗？

**Comment by SpecterShell** [comment]
用来检测软件更新，辅助向[WinGet](https://github.com/microsoft/winget-pkgs)提交新版本的。
可以看看<https://github.com/microsoft/winget-pkgs/tree/master/manifests/h/Huawei/HiSuite>和<https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+HiSuite>有没有需要的历史版本。

**Comment by YU-7** [comment]
是作者你自己挑选开源项目然后上传到winget的社区库吗？有和被上传的仓库作者联系吗？

**Reply by SpecterShell** [reply]
这个仓库只负责<https://github.com/microsoft/winget-pkgs>更新软件包，本身没有软件包信息，但确实有不少软件包是我提交或维护的。

理论上不论是开源项目还是闭源项目，只要可以提交到WinGet社区库并且没有其它人在维护的，看到一个就提交一个，不用挑选。

大多数闭源项目（主要是指那些不用登录或付费就能下载的软件，且网站没有设置Referer或User Agent检测）的许可协议允许其他人在保留版权信息和完整性的条件下分发软件；而开源项目大多通过GitHub releases分发二进制，且其许可协议对分发二进制几乎没有设置限制，故不常在提交软件包前和仓库作者联系。

**Reply by YU-7** [reply]
明白了开源协议本身就很开放，但是那些新提交的其实就是你觉得它不错才帮它提交的吧，不过你这边提交了原作者也提交了不是会有问题吗，毕竟你没有联系过。而且不是也有人反应问题如这个issue https://github.com/SpecterShell/Dumplings/issues/19

**Reply by SpecterShell** [reply]
> 明白了开源协议本身就很开放，但是那些新提交的其实就是你觉得它不错才帮它提交的吧，不过你这边提交了原作者也提交了不是会有问题吗，毕竟你没有联系过。而且不是也有人反应问题如这个issue https://github.com/SpecterShell/Dumplings/issues/19

一般情况下如果有原作者还在持续维护的话我就不再提交了。这个是因为没认出来作者+软件包落后了几个版本。😂

**Comment by ttimasdf** [comment]
> 是作者你自己挑选开源项目然后上传到winget的社区库吗？有和被上传的仓库作者联系吗？

我感觉应该都是未获授权的。看脚本很多都是用爬虫和未公开 API 的方法来拿下载 URL。

还有软件原作者来抱怨的 #19 

不过从结果上来看，确实方便大家了，如果没有这个 repo 我觉得 winget 社区得停摆

**Reply by SpecterShell** [reply]
这个repo主要负责更新<https://github.com/microsoft/winget-pkgs>内的软件包，本身不存储任何软件包信息，需要先把软件包清单提交到那个repo才能起效。

所有通过GitHub/GitLab/Gitee releases分发的软件都是通过GitHub/GitLab/Gitee API获取其链接的。

也有其它自动化项目在做这个，例如<https://github.com/vedantmgoyal9/vedantmgoyal9>。

**Reply by YU-7** [reply]
我问一下更新的频率是怎么样的，脚本定时跑吗，这个时间间隔是多少

**Reply by SpecterShell** [reply]
> 我问一下更新的频率是怎么样的，脚本定时跑吗，这个时间间隔是多少

定时的，每小时一次
